collections/_posts/2023-03-08-Variationnal_Dropout_and_the_local_reparameterization_trick.md

```diff
@@ -0,0 +1,243 @@
+They show that the variance of $$L_{\mathcal{D}}^{SGVB}(\phi)$$  can be dominated by the covariances for even moderately large $$M$$ (size of the minibatch).
+
+$$Var \left[ L_{\mathcal{D}}^{SGVB}(\phi) \right]  = N^2 \left( \frac{1}{M} Var \left[ L_i \right] + \frac{M-1}{M} Cov \left[ L_i, L_j \right] \right)$$
+
+$$L_i = log( p(\textbf{y}^i \vert \textbf{x}^i, \textbf{w} = f (\epsilon^i, \phi)))$$ (contribution to the likelihood for the $$i$$-th datapoint in the minibatch)
+
+
+Therefore they introduce the *Local Reparameterization Trick* to solve this issue. 
+
+To do so they propose a new estimator where the aim is to have $$Cov \left[ L_i, L_j \right] = 0$$ so that the variance of the stochastic gradients scales as $$1/M$$.
+
+Exemple taken in the paper :
+> Consider a standard fully connected neural network containing a hidden layer consisting of 1000 neurons. 
+This layer receives an $$M$$ x $$1000$$ input feature matrix $$\textbf{A}$$ from the layer below, which is multiplied by a $$1000$$ x $$1000$$ weight matrix $$\textbf{W}$$, before a nonlinearity is applied, i.e. $$ \textbf{B = AW}$$. 
+We then specify the posterior approximation on the weights to be a fully factorized Gaussian, i.e. $$q_\phi(w_{i,j}) =N (\mu_{i,j}, \sigma_{i,j}^2) \forall w_{i,j} \in \textbf{W}$$, which means the weights are sampled as 
+$$w_{i,j} = \mu_{i,j} + \sigma_{i,j} \epsilon_{i,j}$$, with $$\epsilon_{i,j} \sim  \mathcal{N}(0, 1)$$.
+In this case we could make sure that $$Cov [L_i,L_j]=0 $$ by sampling a separate weight matrix $$\textbf{W}$$.
+
+This approach is not computationally efficient. To make it so they applied another trick based on 
+> Fortunately, the weights (and therefore $$\epsilon$$) only influence the expected log likelihood through the neuron activations $$\textbf{B}$$, which are of much lower dimension. If we can therefore sample the random activations $$\textbf{B}$$ directly, without sampling $$\textbf{W}$$ or $$\epsilon$$, we may obtain an efficient Monte Carlo estimator at a much lower cost.
+For a factorized Gaussian posterior on the weights, the posterior for the activations (conditional on the input $$\textbf{A}$$) is also factorized Gaussian
+
+
+$$q_\phi(w_{i,j}) =N (\mu_{i,j}, \sigma_{i,j}^2) \forall w_{i,j} \in \textbf{W} \Rightarrow q_\phi(b_{m,j}) =N (\gamma_{m,j}, \delta_{m,j}) $$
+
+with 
+
+$$\gamma_{m,j} = \sum_{i=1}^{1000} a_{m,i}\mu_{i,j}$$ and $$\delta_{m,j} = \sum_{i=1}^{1000} a_{m,i}^2\sigma_{i,j}^2$$
+
+
+>Rather than sampling the Gaussian weights and then computing the resulting activations, 
+we may thus sample the activations from their implied Gaussian distribution directly, 
+using $$b_{m,j} = \gamma_{m,j} + \sqrt{\delta_{m,j}}\zeta_{m,j}$$, with $$\zeta_{m,j} \sim  \mathcal{N}(0, 1)$$. 
+
+Here, $$\zeta$$ is an $$M$$ x $$1000$$ matrix, so only $$M$$ thousand random variables are needed to be sample instead of $$M$$ million.
```

Comment:
I would rephrase the definition of the complexity like this:
```suggestion
Here, $$\zeta$$ is an $$M \times 1000$$ matrix, so only $$M \times 1000$$ random variables have to be sampled instead of $$M \times 1\,000\,000$$.
```